lhotse/augmentation/torchaudio.py

```diff
@@ -212,6 +212,174 @@ def reverse_timestamps(
         )
 
 
+class Codec:
+    def __call__(self, samples: np.ndarray) -> np.ndarray:
+        """
+        Apply encoder then decoder.
+
+        To be implemented in derived classes.
+        """
+        raise NotImplementedError
+
+
+class MuLawCodec:
+    def __init__(self):
+        import torchaudio
+
+        self.encoder = torchaudio.transforms.MuLawEncoding()
+        self.decoder = torchaudio.transforms.MuLawDecoding()
+
+    def __call__(self, samples):
+        return self.decoder(self.encoder(samples))
+
+
+from ctypes import CDLL, POINTER, c_int, c_short, c_uint8, c_void_p
+
+LPC10_FRAME_SAMPLES = 180
+LPC10_FRAME_BYTES = 7
+
+
+def libspandsp_api():
+    api = CDLL("libspandsp.so")
+
+    api.lpc10_encode_init.restype = c_void_p
+    api.lpc10_encode_init.argtypes = [c_void_p, c_int]
+
+    api.lpc10_encode.restype = c_int
+    api.lpc10_encode.argtypes = [c_void_p, POINTER(c_uint8), POINTER(c_short), c_int]
+
+    api.lpc10_encode_free.argtypes = [c_void_p]
+
+    api.lpc10_decode_init.restype = c_void_p
+    api.lpc10_decode_init.argtypes = [c_void_p, c_int]
+
+    api.lpc10_decode.restype = c_int
+    api.lpc10_decode.argtypes = [c_void_p, POINTER(c_short), POINTER(c_uint8), c_int]
+
+    api.lpc10_decode_free.argtypes = [c_void_p]
+
+    return api
+
+
+class LPC10Codec:
+    def __init__(self):
+        self.api = libspandsp_api()
+        self.c_data = (c_uint8 * LPC10_FRAME_BYTES)()
+        self.c_samples = (c_short * LPC10_FRAME_SAMPLES)()
+
+    def __call__(self, samples):
+        encoder = self.api.lpc10_encode_init(None, 0)
+        decoder = self.api.lpc10_decode_init(None, 0)
+
+        frames = samples[0].split(LPC10_FRAME_SAMPLES)
+
+        idx = 0
+        out = torch.zeros([1, len(frames) * LPC10_FRAME_SAMPLES])
+
+        for frame in frames:
+
+            samples_int = (frame * 32768).to(torch.int16)
+
+            for i in range(0, samples_int.shape[0]):
+                self.c_samples[i] = samples_int[i]
+
+            for i in range(samples_int.shape[0], LPC10_FRAME_SAMPLES):
+                self.c_samples[i] = 0
+
+            assert (
+                self.api.lpc10_encode(
+                    encoder, self.c_data, self.c_samples, len(self.c_samples)
+                )
+                == LPC10_FRAME_BYTES
+            )
+            assert (
+                self.api.lpc10_decode(
+                    decoder, self.c_samples, self.c_data, LPC10_FRAME_BYTES
+                )
+                == LPC10_FRAME_SAMPLES
+            )
+
+            for i in range(0, LPC10_FRAME_SAMPLES):
+                out[0][idx] = self.c_samples[i]
+                idx = idx + 1
+
+        self.api.lpc10_encode_free(encoder)
+        self.api.lpc10_decode_free(decoder)
+
+        return out / 32768
+
+
+CODECS = {
+    "lpc10": LPC10Codec,
+    "mulaw": MuLawCodec,
+}
+
+
+@dataclass
+class Phone(AudioTransform):
```

Comment:
I suggest calling it `Narrowband` and renaming the methods to `narrowband`. Also upsampling back to original SR should be optional (`restore_orig_sr=True`).